WordDistanceInSpark.scala

```diff
@@ -1 +1,241 @@
+import org.apache.spark.SparkContext
+import org.apache.spark.SparkContext._
+import org.apache.spark.SparkConf
+import org.apache.spark.rdd.RDD
+
+
+object WordDistanceInSpark {
+  // I'm obviously doing something very wrong, because I get the following exception at the end on the run:
+/* ERROR ContextCleaner: Error in cleaning thread
+java.lang.InterruptedException
+  at java.lang.Object.wait(Native Method)
+  at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:143)
+  at org.apache.spark.ContextCleaner$$anonfun$org$apache$spark$ContextCleaner$$keepCleaning$1.apply$mcV$sp(ContextCleaner.scala:146)
+  at org.apache.spark.ContextCleaner$$anonfun$org$apache$spark$ContextCleaner$$keepCleaning$1.apply(ContextCleaner.scala:144)
+  at org.apache.spark.ContextCleaner$$anonfun$org$apache$spark$ContextCleaner$$keepCleaning$1.apply(ContextCleaner.scala:144)
+  at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:1618)
+  at org.apache.spark.ContextCleaner.org$apache$spark$ContextCleaner$$keepCleaning(ContextCleaner.scala:143)
+  at org.apache.spark.ContextCleaner$$anon$3.run(ContextCleaner.scala:65)
+ERROR Utils: Uncaught exception in thread SparkListenerBus
+java.lang.InterruptedException
+  at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:998)
+  at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1304)
+  at java.util.concurrent.Semaphore.acquire(Semaphore.java:312)
+  at org.apache.spark.util.AsynchronousListenerBus$$anon$1$$anonfun$run$1.apply$mcV$sp(AsynchronousListenerBus.scala:62)
+  at org.apache.spark.util.AsynchronousListenerBus$$anon$1$$anonfun$run$1.apply(AsynchronousListenerBus.scala:61)
+  at org.apache.spark.util.AsynchronousListenerBus$$anon$1$$anonfun$run$1.apply(AsynchronousListenerBus.scala:61)
+  at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:1618)
+  at org.apache.spark.util.AsynchronousListenerBus$$anon$1.run(AsynchronousListenerBus.scala:60)*/
+
+
+  
+  
+  
+  type Hit = (Long, String) // (index, the significant word found at index)
+  type SameWordInterval = ((Long,Long),String) // ( (startIdx, endIdx), significant word)
+  
+  /**
+   * Finds the distance between given words in the text, defined as 
+   * minimum amount of other words in between occurrences of the given words.
+   * 
+   * Distance between a word and itself is 0.
+   * If one of the words does not occur in the text, distance is None
+   *  
+   * @param textFilePath
+   * @param word1
+   * @param word2
+   * @param sc sparkContext
+   * @return
+   */
+  def wordDistance(textFilePath:String, word1:String, word2:String)(implicit sc:SparkContext):Option[Long] = {
+    if (word1 == word2) return Some(0)
```

Comment:
Let's call this a 'hard' case worth extra credits. Could: `d("a b b b b b b b a", a , a) == 7` ?